bot/exts/halloween/candy_collection.py

```diff
@@ -197,20 +185,15 @@ async def candy(self, ctx: commands.Context) -> None:
             '\N{SPORTS MEDAL}'
         )
 
-        top_sorted = sorted(self.candy_json['records'], key=lambda k: k.get('record', 0), reverse=True)
+        top_sorted = sorted(((user_id, score) for user_id, score in self.candy_records.items()),
+                            key=lambda x: x[1], reverse=True)
         top_five = top_sorted[:5]
```

Comment:
you can check if `self.candy_records` validates to `True` before sorting and getting the top_five.
This can be a separate function too.
```python
    def generate_leaderboard(self):
        if not self.candy_records:
            return "No candies."
        top_sorted = sorted(
            ((user_id, score) for user_id, score in self.candy_records.items()),
            key=lambda x: x[1],
            reverse=True
        )
        top_five = top_sorted[:5]

        leaderboard = '\n'.join(
            f"{emoji[index]} <@{record[0]}>: {record[1]}"
            for index, record in enumerate(top_five)
        )
        return leaderboard
```

Reply:
Instead of checking the truthness of the entire dict, I think it's better to check it with `top_five` due to the fact that `candy_records` can contain records with 0 score which still evaluates to false and I'd like to drop them out of leader-board.